src/vs/editor/contrib/codeAction/browser/codeActionUi.ts

```diff
@@ -14,13 +15,15 @@ import { ScrollType } from 'vs/editor/common/editorCommon';
 import { CodeActionTriggerType } from 'vs/editor/common/languages';
 import { MessageController } from 'vs/editor/contrib/message/browser/messageController';
 import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
-import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
 import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
-import { CodeActionAutoApply, CodeActionItem, CodeActionSet, CodeActionTrigger } from '../common/types';
+import { CodeActionAutoApply, CodeActionItem, CodeActionTrigger } from '../common/types';
 import { CodeActionsState } from './codeActionModel';
-import { CodeActionShowOptions, CodeActionWidget } from './codeActionWidget';
+import { CodeActionWidget } from './codeActionWidget';
 import { LightBulbWidget } from './lightBulbWidget';
-
+export interface CodeActionSet extends ActionSet<CodeActionItem> {
+	readonly validActions: readonly CodeActionItem[];
+	readonly allActions: readonly CodeActionItem[];
+}
```

Comment:
Did this move from `'../common/types'` because `ActionSet` is in `browser/`? If that's the only reason should we move baseActionWidget.ts to `vs/base/parts/actionwidget/` which could have multiple layers? Thoughts @mjbvz?

Reply:
> Does this need to move into base? It seems to only be used at high levels, not in base

Matt's comment here might invalidate this